staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -143,6 +143,9 @@ func (intstr *IntOrString) Fuzz(c fuzz.Continue) {
 }
 
 func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool) (int, error) {
+	if intOrPercent == nil {
+		return 0, nil
```

Comment:
What was the old behaviour? panic? error?

This func needs godoc.

Reply:
Panic, although when dealing with internal APIs you didn't have that problem b/c usually these fields are non-pointer. Now that we're slowly going towards external APIs I passed in pointer and it failed terribly.

Reply:
I don't care for this kind of value compression.  Doing this makes nil indistinguishable from `0`.  I'd rather return an error and the places you need to tolerate the nil, you can have a `intOrPercentPtrOrDefault(zero)` that you pass for the first value if that's the behavior you need/want